.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,16 @@
+name: pre-commit
+
+on: [pull_request]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # required to grab the history of the PR
+          fetch-depth: 0
+      - uses: actions/setup-python@v3
+      - uses: pre-commit/action@v3.0.0
+        with:
+          extra_args: --color=always --from-ref ${{ github.event.pull_request.base.sha }} --to-ref ${{ github.event.pull_request.head.sha }}
```

Comment:
Does this action use the pre-commit config file from the repo?

Reply:
Yes, it shall basically do the same checks as the pre-commit hook, and only on changed files. As I understand it we cannot force users to actually install the hook. By this check we check that the code is compatible with the hook. In one of the earlier versions I included some dummy blanks (and disabled the hook) and then the check marked the PR as failed as wanted